src/ProjectOrigin.VerifiableEventStore.Tests/Repository/AbstractTransactionRepositoryTests.cs

```diff
@@ -271,7 +300,48 @@ public async Task CanCreateSeriesOfBlocks()
         block3!.Header.PreviousPublicationHash.Should().BeEquivalentTo(SHA256.HashData(pub2.ToByteArray()));
         block3!.TransactionHashes.Should().HaveCount(transactionList3.Count);
         block3!.TransactionHashes.Should().ContainInOrder(transactionList3.Select(x => x.TransactionHash));
-        var pub3 = new ImmutableLog.V1.BlockPublication();
+        var pub3 = new Registry.V1.BlockPublication();
         await Repository.FinalizeBlock(BlockHash.FromHeader(block3!.Header), pub3);
     }
+
+    [Fact]
+    public async Task CanGetBlocks()
+    {
+        // Arrange
+        for (var b = 0; b < 10; b++)
+        {
+            for (var t = 0; t < 10 * (b + 1); t++)
+            {
+                var @event = CreateFakeEvent(Guid.NewGuid(), 0);
+                await Repository.Store(@event);
+            }
+            var block = await Repository.CreateNextBlock();
+            await Repository.FinalizeBlock(BlockHash.FromHeader(block!.Header), new Registry.V1.BlockPublication());
+        }
+
+        // Act
+        var blocks1 = await Repository.GetBlocks(0, 2, false);
+
+        // Assert
+        blocks1.Should().HaveCount(2);
+        blocks1[0].Header.Should().NotBeNull();
+        blocks1[1].Header.Should().NotBeNull();
+
+        blocks1[0].Height.Should().Be(1);
+        blocks1[0].Transactions.Should().BeEmpty();
+        blocks1[1].Height.Should().Be(2);
+        blocks1[0].Transactions.Should().BeEmpty();
+
+        // Act
+        var blocks2 = await Repository.GetBlocks(6, 2, true);
+        blocks2.Should().HaveCount(2);
+        blocks2[0].Header.Should().NotBeNull();
+        blocks1[1].Header.Should().NotBeNull();
+
+        blocks2[0].Height.Should().Be(7);
+        blocks2[0].Transactions.Should().HaveCount(70);
+        blocks2[1].Height.Should().Be(8);
+        blocks2[1].Transactions.Should().HaveCount(80);
```

Comment:
Consider splitting the assertion of the different blocks up, even further. First assert upon everything that is blocks1[0] related, then do the same for blocks1[1], then blocks2[0], and lastly blocks2[1]. I see tests as trying to tell a story, and doing it that way would make it more understandable for me. 

Suggestion:

```
[Fact]
public async Task BlockRepository_CanRetrieveBlocks_WithExpectedContents()
{
    // Arrange: Create a series of blocks, each with a progressively increasing number of events
    const int totalBlocksToCreate = 10;
    for (int currentBlockNumber = 0; currentBlockNumber < totalBlocksToCreate; currentBlockNumber++)
    {
        int eventsInCurrentBlock = 10 * (currentBlockNumber + 1);
        for (int eventIndex = 0; eventIndex < eventsInCurrentBlock; eventIndex++)
        {
            var newEvent = CreateFakeEvent(Guid.NewGuid(), 0);
            await Repository.Store(newEvent);
        }
        var createdBlock = await Repository.CreateNextBlock();
        await Repository.FinalizeBlock(BlockHash.FromHeader(createdBlock.Header), new Registry.V1.BlockPublication());
    }

    // Act: Retrieve the first two blocks and verify their headers and contents
    var firstTwoBlocks = await Repository.GetBlocks(startIndex: 0, count: 2, includeTransactions: false);

    // Assert: Check that the first two blocks have been retrieved correctly
    firstTwoBlocks.Should().HaveCount(2);

    // Verification for the first two blocks
    firstTwoBlocks[0].Header.Should().NotBeNull();
    firstTwoBlocks[0].Height.Should().Be(1);
    firstTwoBlocks[0].Transactions.Should().BeEmpty();

    firstTwoBlocks[1].Header.Should().NotBeNull();
    firstTwoBlocks[1].Height.Should().Be(2);
    firstTwoBlocks[1].Transactions.Should().BeEmpty();

    // Act: Retrieve two blocks starting from the seventh block, including transactions
    var twoBlocksFromSeventh = await Repository.GetBlocks(startIndex: 6, count: 2, includeTransactions: true);

    // Assert: Check that the blocks have the expected transactions based on their height
    twoBlocksFromSeventh.Should().HaveCount(2);

    // Verification for blocks starting from the seventh
    twoBlocksFromSeventh[0].Header.Should().NotBeNull();
    twoBlocksFromSeventh[0].Height.Should().Be(7);
    twoBlocksFromSeventh[0].Transactions.Should().HaveCount(70);

    twoBlocksFromSeventh[1].Header.Should().NotBeNull();
    twoBlocksFromSeventh[1].Height.Should().Be(8);
    twoBlocksFromSeventh[1].Transactions.Should().HaveCount(80);
}
```